### ISSUE TITLE
Gradient accumulation using Horovod's `backward_passes_per_step` 

### ISSUE THREAD

**Issue by ananthsub** [PyTorchLightning/pytorch-lightning#11732]
## 🚀 Feature

Support this feature for Horovod users with Lightning

### Motivation

Consistency of training with different strategies

### Pitch

Horovod supports gradient accumulation as described here: https://horovod.readthedocs.io/en/stable/adasum_user_guide_include.html#pytorch

Similar to how Deepspeed initializes this:

- Add the property to Horovod: https://github.com/PyTorchLightning/pytorch-lightning/blob/4d72110b51525296eac16454e7b34e2c1b60cb7e/pytorch_lightning/strategies/deepspeed.py#L584-L587

Inside of the `setup` function: https://github.com/PyTorchLightning/pytorch-lightning/blob/4d72110b51525296eac16454e7b34e2c1b60cb7e/pytorch_lightning/strategies/horovod.py#L79

- Verify that a constant accumulation schedule is used: https://github.com/PyTorchLightning/pytorch-lightning/blob/4d72110b51525296eac16454e7b34e2c1b60cb7e/pytorch_lightning/strategies/deepspeed.py#L433-L438
- Then extract the value from `trainer.accumulate_grad_batches` and pass this to the distributed optimizer inside of `_wrap_optimizers`:
https://github.com/PyTorchLightning/pytorch-lightning/blob/4d72110b51525296eac16454e7b34e2c1b60cb7e/pytorch_lightning/strategies/horovod.py#L184-L191

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @awaelchli

**Comment by krshrimali** [comment]
Hi, @ananthsub - Thanks for creating the issue. As a start to contributing to the PyTorch Lightning framework, I wish to fix this. Is it okay if I create a PR following your guidelines in this issue?

cc: @Borda @awaelchli 

P.S: I first tried to see if I can fix this, and since the changes looked straightforward (thanks for the detailed issue), the PR is ready from my end. I'll create one, but please let me know if someone else wished to fix this instead, and I am happy to close my PR.
